### ISSUE TITLE
Impact & applicability of 3.13 Use CSS Preference And Media Queries

### ISSUE THREAD

**Issue by thibaudcolas** [w3c/sustyweb#38]
👋 I thought I’d start sharing my feedback with a specific guideline that I find great but could use improvement because in my opinion it mixes too many things.

> ## [3.13 Use CSS Preference And Media Queries](https://w3c.github.io/sustyweb/#use-css-preference-and-media-queries)
> Website emissions can be generated in numerous ways, some of the latest which can be controlled are through CSS preference queries. By offering the ability to stop animation, remove colors, give a print friendly format, adjust to the available lighting or even offer a less bandwidth hungry version of a page (based on visitor demand), we can provide a less impactful journey.

> ### Success Criterion - CSS Preference Queries
> Account for light-level, monochrome, prefers-contrast, prefers-color-scheme, prefers-reduced-data, prefers-reduced-motion, print & scripting CSS preferences.

I see the following problems here:

- The guideline describes this being about website emissions. For lots of the media queries here, I would argue the primary benefit isn’t reducing emissions, and putting this all under one SC might not be the right approach.
- Recommending use of specific media queries feels more like a technique than a guideline to me (the guideline should be "adapt to user preferences", not "use the following media queries").
- Quite a few of the proposed preference queries aren’t relevant in the real world.
- There are privacy (browser fingerprinting) concerns with lots of those, that should be acknowledged in guidelines about sustainability.

---

## Media query by media query

- `light-level` – [is no longer a proposed standard](https://www.w3.org/TR/mediaqueries-5/#changes-since-2020-07-15) (and has never been implemented anywhere AFAIK).
- `monochrome` – is a thing, but I’d argue it’s too niche / not appropriate enough to warrant mention in a general "preference queries" guideline. It’s really more about device capabilities like `any-pointer` – detecting an B&W e-ink screen or monochrome printer to adapt the content, rather than a user preference.
- `prefers-contrast` – has nothing to do with energy usage / emissions
- `prefers-color-scheme` – helps with energy usage but that’s only one of the elements.
- [`prefers-reduced-data`](https://www.w3.org/TR/mediaqueries-5/#prefers-reduced-data) – is not implemented anywhere due to the browser fingerprinting concerns.
- `prefers-reduced-motion` – helps with energy usage but that’s only one of the elements.
- `print & scripting` – I don’t know what "scripting" refers to.

## Benefits

I also find the wording of the benefits could be improved:

> **Environmental**: You can allow visitor's to disable animation & video (processor heavy), color (OLED and certain colors are power hungry), data heavy code (are bandwidth hungry), and reduce ink & paper usage.

- The standardised intention of `prefers-reduced-motion` is to _minimize_ animation and motion. It can be implemented to fully disable it but I’d recommend those guidelines follow the intent of the CSS spec.
- _color (OLED and certain colors are power hungry)_ this reads as "allow visitors to disable color". I don’t think that’s what `monochrome` does but could be wrong. I also believe OLED energy use is generally lower than LCD, which is the opposite of what’s stated here.
- _data heavy code (are bandwidth hungry)_ – not a benefit since `prefers-reduced-data` doesn’t actually work anywhere.

> **Social Equity**: Media queries don't tell individuals how to experience the web, they follow preferences.

I don’t really understand this point. "Media queries" generally aren’t about user preferences, only some are. Even the ones that aren’t about preferences don’t "tell individuals how to experience the web", they’re generally there to adapt the ""web"" to the user’s device capabilities. And lots of the preference queries discussed raise privacy concerns specifically because they could allow targeting people with low income.

> **Accessibility**: Contrast, less motion, and dark mode may assist the visually impaired.

Could be just me but I’d say those things "will" assist (for the people who choose to use them).

> **Performance**: Reduced data will speed up a website, a benefit to avoiding visitor's data plans.

Again reduced data isn’t really a thing.

> **Economic**: Print stylesheets can save visitor's ink and paper costs (applies to staff on Intranet).

- We’re talking about a print media query here, not print stylesheets. Perhaps a sign those points about "print" should be moved to a separate "print" guideline, that covers both stylesheets and media queries, with the two options relegated to implementation technqiues.
- _(applies to staff on Intranet)_ – do we need to say this? Print stylesheets are universally useful.

> **Conversion**: User preferences make an interface friendlier, encouraging repeat visitor's.

Perhaps also a sign that this guideline would be better as a "user preferences" guideline, with media queries only as a technique. I agree with this point but that could be said of any user preference, so it’s not particularly strong here.

## Proposed changes

Based on the above, I’d recommend to:

- Remove all queries that aren’t currently usable for real-world projects
- Move points about print to a different guideline (for example [2.23 Reduce The Impact Of Downloadable Or Physical Document](https://w3c.github.io/sustyweb/#reduce-the-impact-of-downloadable-or-physical-documents)
- Possibly turn this into a more generic UX guideline about user preferences, rather than a web development guideline about certain media queries. Said guideline could have SCs for certain preferences that have specific media queries available.
- Rewrite the description / split into multiple SCs so it’s clear from the get-go this isn’t just about emissions (rather than having to refer to the benefits)
- Review the privacy concerns associated with those media queries and acknowledge them in the guidelines.

**Comment by AlexDawsonUK** [comment]
Hey @thibaudcolas

I agree with many of your points, regarding your proposed changes I'll summarise:

- I agree it would be better if the guideline were framed as Sustainability benefits as opposed to Website emissions as Media queries are providing benefits not just (as mentioned with print) ecological benefits, but also other benefits which can assist the visitors experience and in doing so either reduce the time wasted on the site (and as a by-product reduce emissions) or have a net benefit effect on hardware performance (again reducing emissions as a by-product).
- light-level will be removed as it's been dropped, however it was included because there was a use-case for allowing automatic dimming of the screen depending on the available ambiant light (as displays use less power if they are on a lower brightness setting, efficiency savings could be made). Regarding the others, each of them were chosen because of their ability to have a sustainability impact (directly or indirectly). Regarding media queries which can't be used (prefers-reduced-data), browser vendors are discussing the best method of providing sustainability savings to consumers, and that media query is among them, we felt it was important to include it as a non-breaking future option.
- I also agree that having separate success criteria would help better describe the conditions of use for each of the media queries, so this will be done.
- I will be sure to add the fingerprinting concerns and make changes based on the spots you've found in the benefits section (above).

**Comment by AlexDawsonUK** [comment]
All of the above has been added into the next draft, including each of the bugs you found and the fingerprinting material plus supporting links to add weight to it. The only adjustment I've made is to the third point made above.

Rather than having a Success Criteria for each media query, the whole guideline and existing Success Criteria have been re-written to better describe the reasoning and justification. The benefits section has also been greatly expanded to describe the mechanics behind why each media / preference query will give said benefit to the visitor.

This should hopefully read better on balance as it leaves the debate for inclusion where it belongs.

The content will appear once the next draft is published!
